utils/src/test/scala/za/co/absa/enceladus/utils/types/DefaultsByFormatSuite.scala

```diff
@@ -23,22 +23,24 @@ class DefaultsByFormatSuite extends AnyFunSuite {
 
   private val customConfig = new ConfigReader(
     ConfigFactory.empty()
-      .withValue("defaultDateTimeZone", ConfigValueFactory.fromAnyRef("PST"))
-      .withValue("defaultDateTimeZone-csv", ConfigValueFactory.fromAnyRef("JST"))
-      .withValue("defaultDateTimeZone-parquet", ConfigValueFactory.fromAnyRef("Gibberish"))
-      .withValue("defaultTimestampTimeZone", ConfigValueFactory.fromAnyRef("CET"))
-      .withValue("defaultTimestampTimeZone-xml", ConfigValueFactory.fromAnyRef("Africa/Johannesburg"))
-      .withValue("defaultTimestampTimeZone-json", ConfigValueFactory.fromAnyRef("WrongTimeZone"))
+      .withValue("defaultTimestampTimeZone", ConfigValueFactory.fromAnyRef("UTC"))
+      .withValue("defaultDateTimeZone", ConfigValueFactory.fromAnyRef("UTC"))
+      .withValue("enceladus.defaultDateTimeZone.default", ConfigValueFactory.fromAnyRef("PST"))
+      .withValue("enceladus.defaultDateTimeZone.csv", ConfigValueFactory.fromAnyRef("JST"))
+      .withValue("enceladus.defaultDateTimeZone.parquet", ConfigValueFactory.fromAnyRef("Gibberish"))
+//      .withValue("enceladus.defaultTimestampTimeZone.default", ConfigValueFactory.fromAnyRef("CET"))
+//      .withValue("enceladus.defaultTimestampTimeZone.xml", ConfigValueFactory.fromAnyRef("Africa/Johannesburg"))
+      .withValue("enceladus.defaultTimestampTimeZone.json", ConfigValueFactory.fromAnyRef("WrongTimeZone"))
   )
 
 
   test("Format specific timestamp time zone override exists") {
-    val default = new DefaultsByFormat("xml", config = customConfig)
+    val default = new DefaultsByFormat("xml")
     assert(default.getDefaultTimestampTimeZone.contains("Africa/Johannesburg"))
   }
 
   test("Format specific timestamp time zone override does not exists") {
-    val default = new DefaultsByFormat("txt", config = customConfig)
+    val default = new DefaultsByFormat("txt")
```

Comment:
I find the test a bit hard to read (hard to orient in). There are couple of aspects I think could be improved:
 - since the date-TZ processing and timestamp-TZ processing logic is not necessarily connected, their tests can be separated, thus easier to reason about
 - there are some obsolete fallback values in the config - a test should exists for them being used when not overshadowed by `enceladus.*.default` (case of `defaultTimestampTimeZone`).

So in the end, my suggestion is this (in this code, I have left the commented out timestamp-TZ config values from test-config):
```scala
private val customTimestampConfig = new ConfigReader(
    ConfigFactory.empty()
      .withValue("defaultTimestampTimeZone", ConfigValueFactory.fromAnyRef("UTC")) // fallback to "obsolete"
      .withValue("enceladus.defaultTimestampTimeZone.json", ConfigValueFactory.fromAnyRef("WrongTimeZone"))
    // this is what the default test-config contains:
    //      .withValue("enceladus.defaultTimestampTimeZone.default", ConfigValueFactory.fromAnyRef("CET"))
    //      .withValue("enceladus.defaultTimestampTimeZone.xml", ConfigValueFactory.fromAnyRef("Africa/Johannesburg"))

  )

  test("Format specific timestamp time zone override exists") {
    val default = new DefaultsByFormat("xml")
    assert(default.getDefaultTimestampTimeZone.contains("Africa/Johannesburg"))
  }

  test("Format specific timestamp time zone override does not exists") {
    val default = new DefaultsByFormat("txt")
    assert(default.getDefaultTimestampTimeZone.contains("CET"))
  }

  test("Format specific timestamp zone fallbacks to obsolete") {
    val defaults = new DefaultsByFormat("xml", config = customTimestampConfig)
    assert(defaults.getDefaultTimestampTimeZone.contains("UTC"))
  }

  test("Format specific date time zone override is not a valid time zone id") {
    intercept[IllegalStateException] {
      new DefaultsByFormat("json", config = customTimestampConfig)
    }
  }

  private val customDateConfig = new ConfigReader(
    ConfigFactory.empty()
      .withValue("defaultDateTimeZone", ConfigValueFactory.fromAnyRef("UTC")) // fallback to "obsolete"
      .withValue("enceladus.defaultDateTimeZone.default", ConfigValueFactory.fromAnyRef("PST"))
      .withValue("enceladus.defaultDateTimeZone.csv", ConfigValueFactory.fromAnyRef("JST"))
      .withValue("enceladus.defaultDateTimeZone.parquet", ConfigValueFactory.fromAnyRef("Gibberish"))
  )

  test("Date time zone does not exist at all") {
    val default = new DefaultsByFormat("testFormat")
    assert(default.getDefaultDateTimeZone.isEmpty)
  }

  test("Format specific date time zone override exists") {
    val defaults = new DefaultsByFormat("csv", config = customDateConfig)
    assert(defaults.getDefaultDateTimeZone.contains("JST"))
  }

  test("Format specific date time zone override does not exists") {
    val defaults = new DefaultsByFormat("testFormat", config = customDateConfig)
    assert(defaults.getDefaultDateTimeZone.contains("PST"))
  }


  test("Format specific date zone override is not a valid time zone id") {
    intercept[IllegalStateException] {
      new DefaultsByFormat("parquet", config = customDateConfig)
    }
  }



```